File: Zombie_Dash/core/src/com/nopalsoft/zombiedash/Achievements.java
package com.nopalsoft.zombiedash;

import com.badlogic.gdx.Gdx;

public class Achievements {

    static boolean didInit = false;


    static String firstStep, zombieKiller, zombieHunter, zombieSlayer, zombieDestroyer, youCanNotHitMe;
    private static boolean doneFristStep, doneZombieYouCanNotHitMe;

    public static void init(MainZombieDash game) {

        firstStep = "20640";
        zombieKiller = "20642";
        zombieHunter = "20644";
        zombieSlayer = "20646";
        zombieDestroyer = "20648";
        youCanNotHitMe = "20650";
        didInit = true;
    }

    /**
     * Called when u start a new game so u can try to do achievements once more
     */
    public static void tryAgainAchievements() {
        doneFristStep = false;
        doneZombieYouCanNotHitMe = false;
    }

    private static void didInit() {
        if (didInit)
            return;
        Gdx.app.log("Achievements", "You must call first Achievements.init()");
    }

    public static void unlockKilledZombies() {
        didInit();
    }

    public static void distance(int distance, boolean didGetHurt) {
        didInit();
        if (distance > 1 && !doneFristStep) {
            doneFristStep = true;
        } else if (distance > 500 && !doneZombieYouCanNotHitMe && !didGetHurt) {
            doneZombieYouCanNotHitMe = true;
        }
    }
}

File: Zombie_Dash/core/src/com/nopalsoft/zombiedash/Achievements.kt
package com.nopalsoft.zombiedash

import com.badlogic.gdx.Gdx

object Achievements {
    var didInit: Boolean = false


    var firstStep: String? = null
    var zombieKiller: String? = null
    var zombieHunter: String? = null
    var zombieSlayer: String? = null
    var zombieDestroyer: String? = null
    var youCanNotHitMe: String? = null
    private var doneFristStep = false
    private var doneZombieYouCanNotHitMe = false

    fun init() {
        firstStep = "20640"
        zombieKiller = "20642"
        zombieHunter = "20644"
        zombieSlayer = "20646"
        zombieDestroyer = "20648"
        youCanNotHitMe = "20650"
        didInit = true
    }

    /**
     * Called when u start a new game so u can try to do achievements once more
     */
    fun tryAgainAchievements() {
        doneFristStep = false
        doneZombieYouCanNotHitMe = false
    }

    private fun didInit() {
        if (didInit) return
        Gdx.app.log("Achievements", "You must call first Achievements.init()")
    }

    fun unlockKilledZombies() {
        didInit()
    }

    fun distance(distance: Int, didGetHurt: Boolean) {
        didInit()
        if (distance > 1 && !doneFristStep) {
            doneFristStep = true
        } else if (distance > 500 && !doneZombieYouCanNotHitMe && !didGetHurt) {
            doneZombieYouCanNotHitMe = true
        }
    }
}